sphinx/transforms/references.py

```diff
@@ -0,0 +1,30 @@
+# -*- coding: utf-8 -*-
+"""
+    sphinx.transforms.references
+    ~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+
+    Docutils transforms used by Sphinx.
+
+    :copyright: Copyright 2007-2018 by the Sphinx team, see AUTHORS.
+    :license: BSD, see LICENSE for details.
+"""
+
+from docutils import nodes
+from docutils.transforms.references import Substitutions
+
+from sphinx.transforms import SphinxTransform
+
+
+class SubstitutionDefinitionsRemover(SphinxTransform):
+    """Remove ``substitution_definition node from doctrees.
+
+    .. note:: In Sphinx-1.7, this transform is only used in LaTeX builder.
```

Comment:
I added this as local transform for LaTeX builder. But `substituion_definition` node is not used in all of builders. So I'll move this to normal transform which is invoked in reading phase, in master branch.